lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runtime/azbatch/util/batch/AzureBatchHelper.java

```diff
@@ -90,19 +111,47 @@ public void submitJob(final String applicationId, final String storageContainerS
         .withName(SharedAccessSignatureCloudBlobClientProvider.AZURE_STORAGE_CONTAINER_SAS_TOKEN_ENV)
         .withValue(storageContainerSAS);
 
+
+    String portMappings = "";
+
+    Iterator<Integer> iterator = this.portProvider.iterator();
+    while (iterator.hasNext()) {
+      Integer port = iterator.next();
+      System.out.println("iter port is " + port);
+      portMappings += String.format("-p %d:%d ", port, port);
+    }
+
+    TaskContainerSettings containerSettings = null;
+    JobPreparationTask jobPreparationTask = null;
+    if (this.containerRegistry != null) {
+      containerSettings = new TaskContainerSettings()
+          .withRegistry(this.containerRegistry)
+          .withImageName(this.containerImageName)
+          .withContainerRunOptions("-dit --env HOST_IP_ADDR_PATH=$AZ_BATCH_JOB_PREP_DIR/hostip.txt " + portMappings);
+      String captureIpAddressCommandLine =
```

Comment:
>String captureIpAddressCommandLine = [](start = 6, length = 36)

command should be const. What about windows? This command needs to conditional on the OS.